bash/bldjar

```diff
@@ -1,53 +1,38 @@
 #!/bin/bash
 
 # bldjar: rebuild z390.jar from src using current installed JDK
-#
-# Note: must be run from z390/bash subdirectory
 
-read -s -p "Press Enter to reuild z390.jar from src using current installed JDK";
-echo "";
+set -e              # exit on all errors
+cd $(dirname $0)    # set to directory of script
 
-# Windows BLDJAR.BAT copied src *.java and Z390.MAN to jar; not done here
+mkdir -p "../jar"   # create folder if does not exist
 
-# ensure destination directory exists; exit if not
-if [ ! -d "../jar" ]; then
-    echo "Error: ../jar directory does not exist"
-    exit 1
-fi
 # empty destination directory; cd to it
-rm ../jar/*
+rm -f ../jar/*
 cd ../jar
+
 # compile z390 java sources to class vm object code
 echo "Compiling ../src/*.java into jar"
 javac -d . -g:none ../src/*.java
 ec=$?
 echo "The exit code of the javac command is $ec"
-# pause to allow the user to terminate
-read -s -p "Press Enter to continue; press Ctrl-C to exit"
-echo ""
-echo ""
+
+# create z390.properties file
+../bash/relver $1
+
```

Comment:
The bash/relver script must be run in a git environment in order to produce a valid z390.properties file.

It is still my opinion that bldjar should not require a git environment. A z390 user should not be required to have git in order to build z390.

Why can't the z390.properties file be built in a git environment, becoming a z390 file that already exists at the point bldjar is run? It should be included in the two "release" zip files.

Reply:
Just to be clear, the properties file is embedded in the jar so is not required for run, only build. In this case, inclusion in the distribution is not required unless you need to build. And the dist is not for building, that's what the source is for.

Reply:
Also the build works without Git.. you need to use "force" param to ignore the error.. I have enhanced the build scripts to check for git.

Reply:
I know that the z390.properties is embedded in the z390.jar file and I know that tz390.java gets the version value at runtime by reading the embedded z390.properties file.

I also understand that, DURING DEVELOPMENT, the version value is in a state of flux so care must be taken with its use. However, when the "release" zip files ("build" and "source") are created, the value is frozen. In order to see the version value without running z390, one can extract the z390.properties file from the z390.jar file contained in the "build" zip file.

I will yield on requiring the z390.properties file being in the "build" zip file. However, it is my opinion that it should be a file in the "source" zip file.

There is no need to continue discussing this issue. Just do what you think is best.